pkg/storage/etcd/etcd_watcher.go

```diff
@@ -398,7 +398,9 @@ func (w *etcdWatcher) sendModify(res *etcd.Response) {
 	var oldObj runtime.Object
 	if res.PrevNode != nil && res.PrevNode.Value != "" {
 		// Ignore problems reading the old object.
-		if oldObj, err = w.decodeObject(res.PrevNode); err == nil {
+		node := *res.PrevNode
+		node.ModifiedIndex = res.Node.ModifiedIndex
```

Comment:
Currently, the old resource version is set in oldObj, which will be returned only in [Deleted event](https://github.com/kubernetes/kubernetes/blob/995f022808d1407897a5b23373484ac683534390/pkg/storage/etcd/etcd_watcher.go#L419-L423).